rpm/brave-keyring.spec

```diff
@@ -8,43 +8,35 @@ URL:        https://www.brave.com/
 Source0:    ./brave-keyring-source.tar.gz
 BuildArch:  noarch
 
-Requires:   at
-
 %description
 The Brave keyring setup installs the keyring files necessary for validating
-packages. In the future it will install the yum.repos.d repository for for
-fetching the packages.
+packages.
 
 %prep
 %setup -q
 
-
-%build
-
-
 %install
 mkdir -p %{buildroot}/etc/pki/rpm-gpg
-mkdir -p %{buildroot}/etc/cron.daily
+mkdir -p %{buildroot}%{_libexecdir}
 mkdir -p %{buildroot}/usr/lib/sysctl.d/
 install -m 644 etc/pki/rpm-gpg/RPM-GPG-KEY-brave -t %{buildroot}/etc/pki/rpm-gpg/
 install -m 644 etc/pki/rpm-gpg/RPM-GPG-KEY-brave-beta -t %{buildroot}/etc/pki/rpm-gpg/
 install -m 644 etc/pki/rpm-gpg/RPM-GPG-KEY-brave-nightly -t %{buildroot}/etc/pki/rpm-gpg/
-install -m 755 etc/cron.daily/brave-key-updater -t %{buildroot}/etc/cron.daily/
+install -m 755 bin/brave-key-updater -t %{buildroot}%{_libexecdir}/
 install -m 644 usr/lib/sysctl.d/53-brave.conf -t %{buildroot}/usr/lib/sysctl.d/
-
 mkdir -p %{buildroot}/etc/yum.repos.d
 
 
 %files
 /etc/pki/rpm-gpg/RPM-GPG-KEY-brave
 /etc/pki/rpm-gpg/RPM-GPG-KEY-brave-beta
 /etc/pki/rpm-gpg/RPM-GPG-KEY-brave-nightly
-/etc/cron.daily/brave-key-updater
+%{_libexecdir}/brave-key-updater
 /usr/lib/sysctl.d/53-brave.conf
 
 %post
-service atd start
-echo "sh /etc/cron.daily/brave-key-updater" | at now + 2 minute > /dev/null 2>&1
+timeout 10m sh -c "while ! sh %{_libexecdir}/brave-key-updater >/dev/null 2>&1; do sleep 2; done" &
```

Comment:
Why do we need the loop?

Reply:
We cannot import the keys while the RPM lock is active. That's why we were scheduling the script with the `at` command. The proposed approach involves executing a detached loop that will keep trying in the background (for up to 10 minutes) until the lock is released and the keys are imported. This approach relies solely on `coreutils`.